controllers/kyma_controller_test.go

```diff
@@ -262,3 +279,27 @@ func updateModuleTemplateSpecData(kymaName, valueUpdated string) func() error {
 		return nil
 	}
 }
+
+func CheckKymaConditions(ctx context.Context, kcpClient client.Client, kymaName string,
+	requiredConditions []v1beta1.KymaConditionType) func() bool {
+	return func() bool {
+		kymaFromCluster, err := GetKyma(ctx, kcpClient, kymaName, "")
+		if err != nil || len(kymaFromCluster.Status.Conditions) != len(requiredConditions) {
+			return false
+		}
+
+		for _, conditionType := range requiredConditions {
+			exists := false
```

Comment:
This logic has flaws, what you implement here is just check if all `requiredConditions` is a subset of `Status.Conditions`, but what you want to check is if requiredCondition is same as `Status.Conditions`, you need to prevent unwanted condition not exists.

Take your test as an example, you can't make sure if that `Deprecated` condition has been removed or not.

Reply:
I overlooked the precondition that the length of two conditions will also compare, thanks @LeelaChacha point it out. The logic is no problem then (however the logic can still be improved, for example, use hashmap to save requiredConditions type as key, then iterating `status.conditions` to check if all key exists, that's the normal way to compare if two unsorted arrays are equal)

The actual reason why that `Deprecated` condition can't be detected is due to the condition initialization having problem (this made-up condition never applied to kyma CR actually), this will be fixed in the follow-up commit.